src/Compatibility/Microsoft.DotNet.ApiCompatibility/AssemblySymbolLoader.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.DotNet.ApiCompatibility
+{
+    public class AssemblySymbolLoader
+    {
+        private readonly HashSet<string> _referenceSearchPaths = new();
+        private readonly List<string> _warnings = new();
+        private readonly Dictionary<string, MetadataReference> _loadedAssemblies;
+        private readonly bool _resolveReferences;
+        private CSharpCompilation _cSharpCompilation;
+
+        public AssemblySymbolLoader(string assemblyName = "", bool resolveAssemblyReferences = false)
+        {
+            if (string.IsNullOrEmpty(assemblyName))
+            {
+                assemblyName = $"AssemblyLoader_{DateTime.Now:MM_dd_yy_HH_mm_ss_FFF}";
+            }
+
+            _loadedAssemblies = new Dictionary<string, MetadataReference>();
+            var compilationOptions = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, nullableContextOptions: NullableContextOptions.Enable);
+            _cSharpCompilation = CSharpCompilation.Create(assemblyName, options: compilationOptions);
+            _resolveReferences = resolveAssemblyReferences;
+        }
+
+        public void AddReferenceSearchDirectories(string paths) => AddReferenceSearchDirectories(SplitPaths(paths));
+
+        public void AddReferenceSearchDirectories(IEnumerable<string> paths)
+        {
+            if (paths == null)
+            {
+                throw new ArgumentNullException(nameof(paths));
+            }
+
+            foreach (string path in paths)
+                _referenceSearchPaths.Add(path);
+        }
+
+        public bool HasRoslynDiagnostics(out IEnumerable<Diagnostic> diagnostics)
+        {
+            diagnostics = _cSharpCompilation.GetDiagnostics();
+            return diagnostics.Any();
+        }
+
+        public bool HasLoadWarnings(out IEnumerable<string> warnings)
+        {
+            warnings = _warnings;
+            return _warnings.Count > 0;
+        }
+
+        private static string[] SplitPaths(string paths) =>
+            paths == null ? null : paths.Split(new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
+
+        public IEnumerable<IAssemblySymbol> LoadAssemblies(string paths) => LoadAssemblies(SplitPaths(paths));
+
+        public IEnumerable<IAssemblySymbol> LoadAssemblies(IEnumerable<string> paths)
+        {
+            if (paths == null)
+            {
+                throw new ArgumentNullException(nameof(paths));
+            }
```

Comment:
we should throw if the path doesnt exists.

Reply:
If the path doesn't exist Roslyn will throw for us.